src/Middleware/HttpLogging/src/W3CLoggingMiddleware.cs

```diff
@@ -0,0 +1,264 @@
+            { W3CLoggingFields.UserName, BitOperations.Log2((int)W3CLoggingFields.UserName) },
+            { W3CLoggingFields.ServerName, BitOperations.Log2((int)W3CLoggingFields.ServerName) },
+            { W3CLoggingFields.ServerIpAddress, BitOperations.Log2((int)W3CLoggingFields.ServerIpAddress) },
+            { W3CLoggingFields.ServerPort, BitOperations.Log2((int)W3CLoggingFields.ServerPort) },
+            { W3CLoggingFields.Method, BitOperations.Log2((int)W3CLoggingFields.Method) },
+            { W3CLoggingFields.UriStem, BitOperations.Log2((int)W3CLoggingFields.UriStem) },
+            { W3CLoggingFields.UriQuery, BitOperations.Log2((int)W3CLoggingFields.UriQuery) },
+            { W3CLoggingFields.ProtocolStatus, BitOperations.Log2((int)W3CLoggingFields.ProtocolStatus) },
+            { W3CLoggingFields.TimeTaken, BitOperations.Log2((int)W3CLoggingFields.TimeTaken) },
+            { W3CLoggingFields.ProtocolVersion, BitOperations.Log2((int)W3CLoggingFields.ProtocolVersion) },
+            { W3CLoggingFields.Host, BitOperations.Log2((int)W3CLoggingFields.Host) },
+            { W3CLoggingFields.UserAgent, BitOperations.Log2((int)W3CLoggingFields.UserAgent) },
+            { W3CLoggingFields.Cookie, BitOperations.Log2((int)W3CLoggingFields.Cookie) },
+            { W3CLoggingFields.Referer, BitOperations.Log2((int)W3CLoggingFields.Referer) }
+        };
+
+        internal static readonly int _fieldsLength = _fieldIndices.Count;
+
+        /// <summary>
+        /// Initializes <see cref="W3CLoggingMiddleware" />.
+        /// </summary>
+        /// <param name="next"></param>
+        /// <param name="options"></param>
+        /// <param name="w3cLogger"></param>
+        public W3CLoggingMiddleware(RequestDelegate next, IOptionsMonitor<W3CLoggerOptions> options, W3CLogger w3cLogger)
+        {
+            _next = next ?? throw new ArgumentNullException(nameof(next));
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (w3cLogger == null)
+            {
+                throw new ArgumentNullException(nameof(w3cLogger));
+            }
+
+            _options = options;
+            _w3cLogger = w3cLogger;
+        }
+
+        /// <summary>
+        /// Invokes the <see cref="HttpLoggingMiddleware" />.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>HttpResponseLog.cs
+        public async Task Invoke(HttpContext context)
+        {
+            var options = _options.CurrentValue;
+
+            var elements = new string[_fieldsLength];
+
+            var now = DateTime.Now;
+            if (options.LoggingFields.HasFlag(W3CLoggingFields.Date))
+            {
+                AddToList(elements, W3CLoggingFields.Date, now.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture));
```

Comment:
Can this directly reference the field index static/const for `W3CLoggingFields.Date`, e.g. `_dateIndex`.